terracotta/drivers/geotiff_raster_store.py

```diff
@@ -138,8 +137,14 @@ def get_raster_tile(
             reprojection_method=settings.REPROJECTION_METHOD,
             resampling_method=settings.RESAMPLING_METHOD,
             target_crs=self._TARGET_CRS,
-            rio_env_options=self._RIO_ENV_OPTIONS,
+            rio_env_options=self._RIO_ENV_OPTIONS
         )
+        
+        try: 
+            kwargs.update(aws_s3_endpoint=settings.RASTER_AWS_S3_ENDPOINT)
+        except AttributeError:
+            pass
```

Comment:
This should never trigger. I think what you want is this:

```suggestion
        if settings.RASTER_AWS_S3_ENDPOINT is not None:
            kwargs.update(aws_s3_endpoint=settings.RASTER_AWS_S3_ENDPOINT)
```

Reply:
oh wait your right - forgot all of the variables get assigned a default value